ext/standard/basic_functions.stub.php

```diff
@@ -3069,13 +3069,13 @@ function mail(string $to, string $subject, string $message, array|string $additi
 function abs(int|float $num): int|float {}
 
 /** @compile-time-eval */
-function ceil(int|float $num): float {}
+function ceil(int|float $num): int|float {}
 
 /** @compile-time-eval */
-function floor(int|float $num): float {}
+function floor(int|float $num): int|float {}
```

Comment:
Is the point to keep ``float`` return type just for 32bit builds?

Also, ``$num`` to be of type ``int`` doesn't make any sense, as the function is a no-op in this case.

Reply:
Wait... that's already a bug in the stubs.
This needs to be backported to 8.1.

Reply:
> Is the point to keep float return type just for 32bit builds?

flooring 2.5e300 should be legal and the result is only representable as float.

Reply:
This RFC proposes to keep the given type if possible.
* float given -> float returned (guarantied)
* int given -> int returned (except for int overflow on round with precision < 0)